From 646a279bf6b2212fc454265049060a4ad85d2d2a Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Fri, 6 Mar 2020 11:14:49 +0100
Subject: [PATCH 2/4] tqma6ul: configure OTG1_PWR pin early in mba6ul_setup_usb

Play safe and set GPIO to low in any case. PWR should only be enabled,
if USB is actively initialized as host.

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 board/tqc/tqma6ul/tqma6ul_mba6ul.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/board/tqc/tqma6ul/tqma6ul_mba6ul.c b/board/tqc/tqma6ul/tqma6ul_mba6ul.c
index 4d2ff68fcd1..3077d0be598 100644
--- a/board/tqc/tqma6ul/tqma6ul_mba6ul.c
+++ b/board/tqc/tqma6ul/tqma6ul_mba6ul.c
@@ -318,6 +318,9 @@ static void mba6ul_setup_usb(void)
 {
 	imx_iomux_v3_setup_multiple_pads(usb_otg1_pads,
 					 ARRAY_SIZE(usb_otg1_pads));
+
+	gpio_request(USB_OTG1_PWR, "usb-otg1-pwr");
+	gpio_direction_output(USB_OTG1_PWR , 0);
 }
 
 int board_usb_phy_mode(int port)
@@ -379,7 +382,6 @@ int board_ehci_power(int port, int on)
 
 	switch (port) {
 		case 0:
-			gpio_request(USB_OTG1_PWR, "usb-otg1-pwr");
 			if (on) {
 				/* enable usb-otg */
 				gpio_direction_output(USB_OTG1_PWR , 1);
-- 
2.17.1

